internal/controller/replication.storage/volumereplication_controller.go

```diff
@@ -449,6 +449,10 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 					instance.Status.LastSyncBytes = &tb
 				}
 			}
+			// Check for nil, as info might be nil for knownGRPCErrors
+			if info != nil {
+				setReplicationCondition(vr.instance, vr.instance.Spec.DataSource.Kind, info.StatusMessage, info.Status)
```

Comment:
Add a comment saying that these conditions only will be set when the VR is in primary state not in secondary state. later this can be moved outside when we want this condition to be set irrespective of the state.

Reply:
Done